Zend/zend_compile.c

```diff
@@ -2867,8 +2865,20 @@ static void zend_compile_list_assign(
 
 		zend_verify_list_assign_target(var_ast, old_style);
 
-		zend_emit_op(&fetch_result, ZEND_FETCH_LIST, expr_node, &dim_node);
-		zend_emit_assign_znode(var_ast, &fetch_result);
+		opline = zend_emit_op(&fetch_result, ZEND_FETCH_LIST, expr_node, &dim_node);
+		if (elem_ast->attr) {
+			opline->extended_value = ZEND_LIST_MAKE_WRITABLE;
+			prev_op = (opline - 1);
+			do {
+				if (prev_op->opcode != ZEND_FETCH_LIST) {
+					break;
+				}
+				prev_op->extended_value = ZEND_LIST_MAKE_WRITABLE;
+			} while ((prev_op = (prev_op - 1)) != NULL);
```

Comment:
In a similar vein, if `$c = []` in that example, would a notice be generate for access of `$c[0]` or not? One of the variables is a simple read, while the other isn't.